webapp/components/Notification/Notification.vue

```diff
@@ -1,40 +1,81 @@
 <template>
-  <ds-space :class="{ read: notification.read, notification: true }" margin-bottom="x-small">
+  <ds-space :class="{ read: notificationData.read, notification: true }" margin-bottom="x-small">
     <client-only>
       <ds-space margin-bottom="x-small">
-        <user-teaser :user="from.author" :date-time="from.createdAt" />
+        <user-teaser :user="notificationData.triggerer" :date-time="notificationData.createdAt" />
       </ds-space>
-      <ds-text class="reason-text-for-test" color="soft">
-        {{ $t(`notifications.reason.${notification.reason}`) }}
+      <ds-text data-test="reason-text" color="soft">
+        <base-icon
+          :name="notificationData.iconName"
+          v-tooltip="{ content: $t(notificationData.iconTooltip), placement: 'right' }"
+        />
+        {{
+          $t(
+            `notifications.reason.${notificationData.reason}` +
+              notificationData.reasonTranslationExtention,
+          )
+        }}
       </ds-text>
     </client-only>
     <ds-space margin-bottom="x-small" />
-    <nuxt-link
-      class="notification-mention-post"
-      :to="{ name: 'post-id-slug', params, ...hashParam }"
-      @click.native="$emit('read')"
-    >
+    <nuxt-link :to="notificationData.linkTo" @click.native="$emit('read')">
       <ds-space margin-bottom="x-small">
         <ds-card
-          :header="from.title || from.post.title"
+          :header="
+            (notificationData.isPost || notificationData.isComment) && notificationData.title
+          "
           hover
           space="x-small"
           class="notifications-card"
         >
-          <ds-space margin-bottom="x-small" />
-          <div>
-            <span v-if="isComment" class="comment-notification-header">
+          <!-- because of different margin above ds-card content without header property -->
+          <user-teaser
+            v-if="notificationData.isUser"
+            class="notifications-user-teaser"
+            :user="notificationData.user"
+          />
+          <div v-else>
+            <span v-if="notificationData.isComment" class="notification-content-header-text">
               {{ $t(`notifications.comment`) }}:
             </span>
-            {{ from.contentExcerpt | removeHtml }}
+            {{ notificationData.contentExcerpt | removeHtml }}
           </div>
+          <div v-if="notificationData.isReport">
+            <ds-space margin-bottom="x-small" />
+            <span class="notification-content-header-text">
+              {{ $t(`notifications.report.category`) }}:
+            </span>
+            {{ $t('report.reason.category.options.' + notificationData.report.reasonCategory) }}
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-01-23T19:25:40Z" title="Thursday, January 23rd 2020, 8:25:40 pm +01:00">Jan 23, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -1,40 +1,90 @@
 <template>
-  <ds-space :class="{ read: notification.read, notification: true }" margin-bottom="x-small">
+  <ds-space :class="{ read: notificationData.read, notification: true }" margin-bottom="x-small">
     <client-only>
       <ds-space margin-bottom="x-small">
-        <user-teaser :user="from.author" :date-time="from.createdAt" />
+        <user-teaser :user="notificationData.triggerer" :date-time="notificationData.createdAt" />
       </ds-space>
       <ds-text class="reason-text-for-test" color="soft">
-        {{ $t(`notifications.reason.${notification.reason}`) }}
+        <base-icon
+          v-if="notificationData.report"
+          name="balance-scale"
+          v-tooltip="{ content: $t('notifications.report.name'), placement: 'right' }"
+        />
+        <base-icon
+          v-else-if="notificationData.comment"
+          name="comment"
+          v-tooltip="{ content: $t('notifications.comment'), placement: 'right' }"
+        />
+        <base-icon
+          v-else
+          name="bookmark"
+          v-tooltip="{ content: $t('notifications.post'), placement: 'right' }"
+        />
+        {{
+          $t(`notifications.reason.${notificationData.reason}` + notificationData.reasonExtention)
+        }}
       </ds-text>
     </client-only>
     <ds-space margin-bottom="x-small" />
     <nuxt-link
-      class="notification-mention-post"
-      :to="{ name: 'post-id-slug', params, ...hashParam }"
+      class="notification-link-for-test"
+      :to="notificationData.linkTo"
       @click.native="$emit('read')"
     >
       <ds-space margin-bottom="x-small">
         <ds-card
-          :header="from.title || from.post.title"
+          :header="(notificationData.post || notificationData.comment) && notificationData.title"
           hover
           space="x-small"
           class="notifications-card"
         >
-          <ds-space margin-bottom="x-small" />
-          <div>
-            <span v-if="isComment" class="comment-notification-header">
+          <div v-if="notificationData.user">
+            <!-- because of different margin above ds-card content without header property -->
+            <ds-space margin-bottom="small" />
+            <user-teaser :user="notificationData.user" />
+          </div>
+          <div v-else-if="notificationData.contentExcerpt">
+            <span v-if="notificationData.comment" class="notification-content-header-text">
               {{ $t(`notifications.comment`) }}:
             </span>
-            {{ from.contentExcerpt | removeHtml }}
+            {{ notificationData.contentExcerpt | removeHtml }}
           </div>
+          <div v-if="notificationData.report">
```


maybe we could use `methods` like `isComment`, `isPost`, `isReport` like we do elsewhere?

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2020-02-05T11:21:16Z" title="Wednesday, February 5th 2020, 12:21:16 pm +01:00">Feb 5, 2020</time>_

---


I would like to discuss this personal with you. @mattwr18 
And it’s related to the use of the function `extractNotificationDataOfCurrentUser` as well.